### ISSUE TITLE
Make the neovide logo match macos design

### ISSUE THREAD

**Issue by PumpedSardines** [neovide/neovide#1629]
**Is your feature request related to a problem? Please describe.**
The current logo for neovide is good, but since BigSur most app logos are boxed in in a roundedbox. The macos build of neovide should have an icon that matches that.

**Describe the solution you'd like**
Take the currenticon and place it in front of a white / gray / faded box with radius. Nothing fancy

**Additional context**
Neovide compared to some other popular programs macos logo
<img width="217" alt="Capture d’écran 2022-11-07 à 10 29 07" src="https://user-images.githubusercontent.com/58148984/200275365-e6622d6c-26bd-4f0e-a06c-693f0b97fe9c.png">

My (very bad attempt) at creating an example icon
![image](https://user-images.githubusercontent.com/58148984/200276875-a68f9131-c80d-4201-9f27-c348f82874f0.jpg)

**Comment by xarthurx** [comment]
Agree, on Windows it also looks "broken" with those "gear notches":

![image](https://user-images.githubusercontent.com/1921878/228197223-8eecec70-56c9-4f76-b321-d151381d0b2e.png)

**Comment by himhaidar** [comment]
Definitely, it looks pretty inconsistent with the rest of of the macOS look. This [site](https://www.appicon.co) can generate the icon set for macOS in a couple of clicks.

**Comment by fredizzimo** [comment]
We might also have to redesign the whole logo at some point, it's very close the trademarked Rust logo. Even with the current guidelines it's questionable if we can use it https://foundation.rust-lang.org/policies/logo-policy-and-media-guide/#appearing-official%2C-affiliated%2C-or-endorsed. And the Rust foundation is planning more strict guidelines in the future, they already had a draft version, but it was criticized so much that they have to make revisions to it.

**Comment by 9mm** [comment]
This guy @mark2o did an insanely awesome job, and is very congruent with OSX style guide, potentially you could reach out to him and he'd do it.

https://github.com/marc2o/Neovide-macOS-Icon

https://marc2o.github.io/

![](https://github.com/marc2o/Neovide-macOS-Icon/raw/main/neovide-preview.png)

**Comment by vpashkov** [comment]
@9mm I've built Neovide with these icons. It looks amazing: 
![Image](https://github.com/user-attachments/assets/ff495d22-176d-4776-983a-cfdbb5863cf3)

@marc2o, @fredizzimo I can make a PR that will add the new icon. I'm not sure how to replace the icon in the [source.pdf](https://github.com/neovide/neovide/blob/627956990b4d85a2a98a13e0c50da2155d15cd72/assets/dmg-background/source.pdf) and how to attribute Marc's work properly, but with your help, we'll figure it out.

**Comment by fredizzimo** [comment]
That icon unfortunately uses parts of the trademarked rust logo, so that should be avoided. We would like to completely redesign the icon on all platforms, but the progress on that has stalled. 

See the latest discussion here https://discord.com/channels/896132176091955261/1329582044182741094

**Comment by fredizzimo** [comment]
It's in a thread called "retaking the icon development from scratch" in the development channel of our discord https://discord.gg/SjFpZdQys6. The thread was created to allow a faster feedback loop and more natural discussion than what a GitHub discussion allows. 

That icon by danbee was the first one discussed there, and he is the main participant in the discord thread. The main concern with that icon is that it doesn't look too good at small sizes, and also that making it fit with the style of other platforms might be challenging.

**Comment by vpashkov** [comment]
@oschrenk if you have a desktop app installed, you can try this link: discord://discord.com/channels/896132176091955261/1329582044182741094
